contracts/strategies/StrategyDAI3poolV2.sol

```diff
@@ -0,0 +1,306 @@
+// SPDX-License-Identifier: AGPL-3.0
+
+pragma solidity ^0.5.17;
+
+import "@openzeppelinV2/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelinV2/contracts/math/SafeMath.sol";
+import "@openzeppelinV2/contracts/utils/Address.sol";
+import "@openzeppelinV2/contracts/token/ERC20/SafeERC20.sol";
+
+import "../../interfaces/curve/Curve.sol";
+import "../../interfaces/yearn/IController.sol";
+
+interface yvERC20 {
+    function deposit(uint256) external;
+
+    function withdraw(uint256) external;
+
+    function getPricePerFullShare() external view returns (uint256);
+
+    function earn() external;
+}
+
+/*
+
+ A strategy must implement the following calls;
+ 
+ - deposit()
+ - withdraw(address) must exclude any tokens used in the yield - Controller role - withdraw should return to Controller
+ - withdraw(uint) - Controller | Vault role - withdraw should always return to vault
+ - withdrawAll() - Controller | Vault role - withdraw should always return to vault
+ - balanceOf()
+ 
+ Where possible, strategies must remain as immutable as possible, instead of updating variables, we update the contract by linking it in the controller
+ 
+*/
+
+contract StrategyDAI3poolV2 {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    address public constant want = address(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    address public constant _3pool = address(0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7);
+    address public constant _3crv = address(0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490);
+    address public constant y3crv = address(0x9cA85572E6A3EbF24dEDd195623F188735A5179f);
+
+    address public governance;
+    address public controller;
+    address public strategist;
+    address public keeper;
+
+    uint256 public constant DENOMINATOR = 10000;
+    uint256 public treasuryFee = 1000;
+    uint256 public withdrawalFee = 50;
+    uint256 public strategistReward = 1000;
+    uint256 public threshold = 6000;
+    uint256 public slip = 5;
+    uint256 public tank = 0;
+    uint256 public p = 0;
+    uint256 public maxAmount = 1e24;
+
+    modifier isAuthorized() {
+        require(msg.sender == strategist || msg.sender == governance || msg.sender == controller || msg.sender == address(this), "!authorized");
+        _;
+    }
+
+    constructor(address _controller) public {
+        governance = address(0xFEB4acf3df3cDEA7399794D0869ef76A6EfAff52);
+        strategist = msg.sender;
+        keeper = msg.sender;
+        controller = _controller;
+    }
+
+    function getName() external pure returns (string memory) {
+        return "StrategyDAI3pool";
+    }
+
+    function harvest() external {
+        require(msg.sender == keeper || msg.sender == strategist || msg.sender == governance, "!ksg");
+        rebalance();
+        uint256 _want = (IERC20(want).balanceOf(address(this))).sub(tank);
+        if (_want > 0) {
+            if (_want > maxAmount) _want = maxAmount;
```

Comment:
thoughts on adding this for further adjustments?

`
    function setMaxAmount(uint256 _maxAmount) external {
        require(msg.sender == strategist || msg.sender == governance, "!sg");
        maxAmount = _maxAmount;
    }

`

Reply:
nice catch!